go/tools/builders/importcfg.go

```diff
@@ -196,6 +219,49 @@ package with this path is linked.`,
 		depsSeen[arc.packagePath] = arc.importPath
 		fmt.Fprintf(buf, "packagefile %s=%s\n", arc.packagePath, arc.file)
 	}
+
+	// Generate buildinfo if dependencies are provided
+	if cfg.deps != nil && len(cfg.deps) > 0 {
+		buildInfo := BuildInfo{
+			GoVersion: runtime.Version(),
+			Path:      cfg.path,
+			Main:      Module{},
+			Deps:      cfg.deps,
+			Settings:  []BuildSetting{},
+		}
+
+		buildSettingAppend := func(key, value string) {
+			if value != "" {
+				buildInfo.Settings = append(buildInfo.Settings, BuildSetting{Key: key, Value: value})
+			}
+		}
+
+		buildSettingAppend("-compiler", cfg.compiler)
+		buildSettingAppend("-buildmode", cfg.buildMode)
+		if cfg.pgoProfilePath != "" {
+			buildSettingAppend("-pgo", cfg.pgoProfilePath)
```

Comment:
We need to check that these paths aren't absolute. It would be very easy to accidentally destroy hermeticity that way. This also applies to other flags below.